src/components/Input/Input.tsx

```diff
@@ -35,18 +35,34 @@ import styles from './input.module.css'
 export const defaults = {
   ...baseInputDefaults,
   htmlType: Type.Text,
+  valuePropName: 'value',
 }
 
 const DEFAULT_ICON_SIZE = 12 as never
 
 type AddonPosition = 'before' | 'after'
 
-const getAddonDefaultValue = (
-  position: AddonPosition,
-  props?: InputAddonProps
-): Record<string, unknown> => {
-  return props && (props.value !== undefined || props.defaultValue !== undefined)
-    ? { [position]: props.value || props.defaultValue } : {}
+const getAddonDefaultValue = (position: AddonPosition) =>
+  (props?: InputAddonProps) : Record<string, unknown> => {
+    return props && (props.value !== undefined || props.defaultValue !== undefined)
+      ? { [props.name || position]: props.value || props.defaultValue }
+      : {}
+  }
+
+const useInputValue = (
+  value: Record<string, unknown> | string | undefined,
+  valuePropName: string,
+  { before, after }: Record<AddonPosition, InputAddonProps | undefined>
+):
+  [
+    Record<string, unknown>,
+    React.Dispatch<React.SetStateAction<Record<string, unknown>>>
+  ] => {
+  return useState({
+    ...(typeof value === 'object' ? value : { [valuePropName]: value || '' }),
+    ...getAddonDefaultValue('before')(before),
```

Comment:
Why function that returns another function and not directly `getAddonDefaultValue('before', before)`? I miss something?

Reply:
fixed